src/components/ExpandableListItem.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+
+import {ListItem, ListItemText, ListItemIcon, Collapse, List} from '@material-ui/core';
+import { ExpandMore, ExpandLess } from '@material-ui/icons';
+
+import { Field } from 'health-level-seven-parser'
+
+const ExpandableListItem: React.FC<{
+  field: Field;
+  expandableKey: string;
+  expandableClassName: string;
+  expandableOnClick: () => void;
+  nestedClassName: string
+}> = (props) => {
+
+  const [open, setOpen] = React.useState(false);
+
+  const {
+    field,
+    expandableKey,
+    expandableClassName,
+    expandableOnClick,
+    nestedClassName
+  } = props;
+
+  function handleClick() {
+    setOpen(!open);
+  }
+
+  return <div>
+    <ListItem button
+      key={expandableKey}
+      className={expandableClassName}
+      onClick={() => {
+        expandableOnClick();
+        handleClick();
```

Comment:
We can probably just call `setOpen(!open)` here instead of the extra wrapper function.